bridges/otellogr/logsink_test.go

```diff
@@ -0,0 +1,578 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otellogr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/embedded"
+	"go.opentelemetry.io/otel/log/global"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+)
+
+// embeddedLogger is a type alias so the embedded.Logger type doesn't conflict
+// with the Logger method of the recorder when it is embedded.
+type embeddedLogger = embedded.Logger // nolint:unused  // Used below.
+
+// recorder records all [log.Record]s it is ased to emit.
+type recorder struct {
```

Comment:
Can you use https://pkg.go.dev/go.opentelemetry.io/otel/log@v0.1.0-alpha.0.20240409140646-648b40eae158/logtest instead?

You can get it via `go get go.opentelemetry.io/otel/log@v0.1.0-alpha.0.20240409140646-648b40eae158`.